staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -424,3 +437,66 @@ func testSetupWithEtcdServer(t *testing.T, opts ...setupOption) (context.Context
 
 	return ctx, cacher, server, terminate
 }
+
+func testSetupWithEtcdAndCreateWrapper(t *testing.T, opts ...setupOption) (storage.Interface, func()) {
+	var cacher *Cacher
```

Comment:
I was considering this but was put off by the fact that the function injects a `list error` but it seems to have no effect on this tests since we are waiting for the cacher to be ready.